### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by Sarkar**
+ can't go back to splash activity [todo] - create basefragmentactivity that handles common case of back pressed

+ fixes bug when viewer/creator activity wouldn't open after activity was stopped.

@aaustin @shortstuffsushi

### CODE REVIEW

File: src/io/branch/branchster/SplashActivity.java [hunk 1]
```diff
@@ -98,7 +62,47 @@ public void run() {
...
+                    MonsterPreferences prefs = MonsterPreferences.getInstance(getApplicationContext());
+                    Intent i;
+                    if (referringParams.has("monster")) {
+
+                        prefs.setMonsterName(referringParams.getString("monster_name"));
+                        prefs.setFaceIndex(referringParams.getInt("face_index"));
+                        prefs.setBodyIndex(referringParams.getInt("body_index"));
+                        prefs.setColorIndex(referringParams.getInt("color_index"));
+
+                        i = new Intent(getApplicationContext(), MonsterViewerActivity.class);
```
Comment (shortstuffsushi) [inline src/io/branch/branchster/SplashActivity.java]: You set this to the same item in all conditions.

File: src/io/branch/branchster/SplashActivity.java [hunk 2]
```diff
@@ -98,7 +62,47 @@ public void run() {
...
+
+                    } else {
+
+                        if (prefs.getMonsterName() == null) {
```
Comment (shortstuffsushi) [inline src/io/branch/branchster/SplashActivity.java]: Why set it to "" vs null?

Comment (Sarkar) [inline src/io/branch/branchster/SplashActivity.java]: Not sure - I just copied this over from the `onStart` block, but from your feedback let me clean some of this code up

File: build.gradle [hunk 3]
```diff
@@ -30,7 +30,7 @@ dependencies {
     // But ignore the Facebook and Branch *.jar files, as they're only for Eclipse use.
 
     // Branch SDK
-    compile 'io.branch.sdk.android:library:1.3.5'
+    compile 'io.branch.sdk.android:library:1.5.0'
```
Comment (qinweigong) [inline build.gradle]: 1.5.2?

File: src/io/branch/branchster/SplashActivity.java (hunk 1 above)
Comment (Sarkar) [inline src/io/branch/branchster/SplashActivity.java]: ? This gets set to a different class depending on what we determine we have

File: src/io/branch/branchster/SplashActivity.java [hunk 4]
```diff
@@ -98,7 +65,41 @@ public void run() {
 	    }.start();
 
 	}
-	
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        branch = Branch.getInstance(mApplicationContext, getResources().getString(R.string.bnc_app_key));
```
Comment (qinweigong) [inline src/io/branch/branchster/SplashActivity.java]: As an open source demo app, we'd better show user the more formal way to set branch_key (not bnc_app_key) in the strings.xml & manifest.

**Comment by shortstuffsushi** [comment]
Seems k :+1:

**Comment by qinweigong** [comment]
:+1:

**Pull request merged**